packages/nx-plugin/src/executors/autorun/executor.integration.test.ts

```diff
@@ -0,0 +1,127 @@
+// eslint-disable-next-line n/no-sync
+import { execSync } from 'node:child_process';
+import { ExecutorContext } from 'nx/src/config/misc-interfaces';
+import { expect, vi } from 'vitest';
+import type { UploadConfig } from '@code-pushup/models';
+import { globalConfig, persistConfig, uploadConfig } from '../internal/config';
+import { runAutorunExecutor } from './executor';
+import { parseAutorunExecutorOptions } from './utils';
+
+vi.mock('node:child_process', async () => {
+  const actual = await vi.importActual('node:child_process');
+
+  return {
+    ...actual,
+    // eslint-disable-next-line n/no-sync
+    execSync: vi.fn(),
+  };
+});
+
+vi.mock('../internal/config', async () => {
+  const actual: any = await vi.importActual('../internal/config');
+
+  return {
+    ...actual,
+    persistConfig: vi.fn(actual.persistConfig),
+    uploadConfig: vi.fn(actual.uploadConfig),
+    globalConfig: vi.fn(actual.globalConfig),
+  };
+});
+
+const projectName = 'my-lib';
+const context = {
+  projectName,
+  root: '.', // workspaceRoot
+  projectsConfigurations: {
+    projects: {
+      [projectName]: {
+        name: projectName,
+        root: `libs/${projectName}`,
+      },
+    },
+  },
+} as unknown as ExecutorContext;
```

Comment:
Please stop using `as unknown as T`. I've complained about it so many times in your PRs, and I'm not going to stop. But I will stop refactoring it for you, it seems to be counterproductive :disappointed:

Reply:
Sry for that! :(